sdk/servicebus/azure-servicebus/azure/servicebus/_common/auto_lock_renewer.py

```diff
@@ -120,7 +159,26 @@ def _renewable(self, renewable):
             return False
         return True
 
-    def _auto_lock_renew(
+    def _dispatch_worker(self):
+        self._last_activity_timestamp = time.time()
+        while not self._shutdown.is_set() and self._running.is_set():
+            while not self._renew_tasks.empty():
+                renew_task = self._renew_tasks.get()
+                if self._is_max_workers_greater_than_one:
+                    self._executor.submit(self._auto_lock_renew_task, *renew_task)
+                else:
+                    self._auto_lock_renew_task(*renew_task)
+                self._renew_tasks.task_done()
```

Comment:
should this be renew_task.done()? since `self._renew_tasks` is a queue, which doesn't have a `task_done()` method.

Reply:
we're using the `Queue.queue` class which is synchronized queue (thread safe) and has the task_done method.

reference could be found here: https://docs.python.org/3/library/queue.html#queue.Queue.task_done